lib/cli/tests.rs

```diff
@@ -373,3 +372,51 @@ fn should_fail_to_create_transfer_without_account() {
         CliError::Core(Error::DeployMissingSessionAccount)
     ));
 }
+
+#[test]
+fn should_fail_to_create_transfer_with_no_secret_key_while_not_allowing_unsigned_deploy() {
+    let deploy_params = deploy_params_without_secret_key();
+    let payment_params =
+        PaymentStrParams::with_package_hash(PKG_HASH, VERSION, ENTRYPOINT, args_simple(), "", "");
+
+    let transfer_deploy = deploy::new_transfer(
+        "10000",
+        None,
+        "bad public key.",
```

Comment:
I think it would be better to use a valid `PublicKey` here, as we ideally want just the one aspect to be invalid.

As currently written, this test would start failing if we re-order the arg parsing inside `new_transfer` to get `target` before getting `maybe_secret_key`.